src/Raven.Server/Documents/Queries/QueryBuilder.cs

```diff
@@ -415,12 +431,77 @@ private static MethodExpression FindMoreLikeThisExpression(QueryExpression expre
             throw new InvalidQueryException("Unable to understand query", query.QueryText, parameters);
         }
 
-        private static bool IsExact(IndexDefinitionBase indexDefinition, bool exact, QueryFieldName fieldName)
+        private static bool TryUseTime(Index index, string fieldName, object valueFirst, object valueSecond, bool exact, out long ticksFirst, out long ticksSecond)
+        {
+            ticksFirst = -1;
+            ticksSecond = -1;
+
+            if (exact || index == null || valueFirst == null || valueSecond == null || index.Definition.Version < IndexDefinitionBase.IndexVersion.TimeTicks)
+                return false;
+
+            if (index.IndexFieldsPersistence.HasTimeValues(fieldName) && TryGetTime(valueFirst, out ticksFirst) && TryGetTime(valueSecond, out ticksSecond))
+                return true;
+
+            return false;
+        }
+
+        private static bool TryUseTime(Index index, string fieldName, object value, bool exact, out long ticks)
+        {
+            ticks = -1;
+
+            if (exact || index == null || value == null || index.Definition.Version < IndexDefinitionBase.IndexVersion.TimeTicks)
+                return false;
+
+            if (index.IndexFieldsPersistence.HasTimeValues(fieldName) && TryGetTime(value, out ticks))
+                return true;
+
+            return false;
+        }
+
+        private unsafe static bool TryGetTime(object value, out long ticks)
+        {
+            ticks = -1;
+            DateTime dt = default;
+            DateTimeOffset dto = default;
+            LazyStringParser.Result result = LazyStringParser.Result.Failed;
+
+            switch (value)
+            {
+                case LazyStringValue lsv:
+                    result = LazyStringParser.TryParseDateTime(lsv.Buffer, lsv.Size, out dt, out dto);
+                    break;
+                case string valueAsString:
+                    fixed (char* buffer = valueAsString)
+                        result = LazyStringParser.TryParseDateTime(buffer, valueAsString.Length, out dt, out dto);
+                    break;
+                default:
+                    var otherAsString = value.ToString();
+                    fixed (char* buffer = otherAsString)
+                        result = LazyStringParser.TryParseDateTime(buffer, otherAsString.Length, out dt, out dto);
+                    break;
+            }
+
+            switch (result)
+            {
+                case LazyStringParser.Result.Failed:
+                    return false;
+                case LazyStringParser.Result.DateTime:
+                    ticks = dt.Ticks;
+                    return true;
+                case LazyStringParser.Result.DateTimeOffset:
+                    ticks = dto.UtcDateTime.Ticks; // TODO [ppekrol]
```

Comment:
why todo here?

Reply:
I've marked it as todo, because I was not sure how to handle that, but at later stage I did handle this, but forgot to remove the TODOs